lib/nmatrix/lapack.rb

```diff
@@ -1,8 +1,20 @@
 module NumRuby::Linalg
-  def self.inv(obj)
-    if obj.is_a?(NMatrix)
-      return obj.invert
+  def self.inv(matrix)
```

Comment:
I think you could put this in functions. For example:
```ruby
def self.inv(matrix)
  is_a_matrix?(matrix)
  valid_shape?(matrix, 2)
  ...
end

private

def is_a_matrix?(matrix)
  raise("Invalid matrix. Not of type NMatrix.") unless matrix.is_a?(NMatrix)
end

def valid_shape?(matrix, shape)
  raise("Invalid shape of matrix. Should be 2.") unless matrix.dim != shape
end
```
WDYT? 🤔

Reply:
Yeah, seems alright. Functions for these would be a good idea as these are quite frequent. I'll add these once the PR is complete. Thanks.